src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -134,7 +135,7 @@ export class CourtTypesController {
 
   private setCode(name: string, magistratesCourtCode: string, countyCourtCode: string, crownCourtCode: string){
 
-    const regExp = /^[0-9]*$/;
+    const regExp = /^[1-9][0-9]*$/;
 
     switch (name) {
```

Comment:
Hey @Majid-justice - isn't this switch statement (140-152) redundant as the each case does the same thing, just targeting a different parameter? Couldn't this function instead just take the court code and test with regex, then parse if in the correct format?

Reply:
The switch statement to map court type to the correct court code and also check against regex. Even if I take out the regex to a different function still will require a switch to handle mapping.

Reply:
```
private setCode(name: string, courtCode: string){
  const regExp = /^[1-9][0-9]*$/;

  if (name === magistrateCourtType || name === countyCourtType || name === crownCourtType) {
      return regExp.test(courtCode) ? parseInt(courtCode) : null ;
  }
  
  return null
}
```

Reply:
the courtCode is entered in input elements req.body.magistratesCourtCode, req.body.countyCourtCode, req.body.crownCourtCode and the CourtType is selected from the checkbox, so we need to make sure the correct code is assigned to the associated CourtType

using if statements will look like this 

private setCode(name: string, magistratesCourtCode: string, countyCourtCode: string, crownCourtCode: string){

    const regExp = /^[1-9][0-9]{0,8}$/;

    if (name === this.magistrateCourtType || name === this.countyCourtType || name === this.crownCourtType) {

      if(name === this.magistrateCourtType)
      return regExp.test(magistratesCourtCode) ? parseInt(magistratesCourtCode) : null ;

      if(name === this.countyCourtType)
        return regExp.test(countyCourtCode) ? parseInt(countyCourtCode) : null ;

      if(name === this.crownCourtType)
        return regExp.test(crownCourtCode) ? parseInt(crownCourtCode) : null ;
    }
    return null;

    
  }

Reply:
Ohh I get what u mean here now....there are individual court code passing it there

Reply:
Another solution could be separate the validation out which will look like this

 private setCode(name: string, magistratesCourtCode: string, countyCourtCode: string, crownCourtCode: string){
    
    switch (name) {
      case this.magistrateCourtType:
        return this.ValidateCode(magistratesCourtCode);

      case this.countyCourtType:
        return this.ValidateCode(countyCourtCode);

      case this.crownCourtType:
        return this.ValidateCode(crownCourtCode);

      default:
        return null;
    }
  }
  
  private ValidateCode(code: string)
  {
    const regExp = /^[1-9][0-9]{0,8}$/;
    return regExp.test(code) ? parseInt(code) : null ;
    
  }

Reply:
I think we need a bigger change if we need to change this logic here. The following CourtTypeItem is for an individual court type but it has fields for all court codes. I thought when we receive the object back from the API, the court type is already linked to the court code. Might need a bit more discussion on this.

export interface CourtTypeItem {
  value: string,
  text: string,
  magistrate: boolean,
  county: boolean,
  crown: boolean,
  checked: boolean,
  code: number
}

Reply:
The CourtTypeItem is created to match the Item of checkbox macro so we can dynamically display the Court types, also the 3 booleans make it clear which input element should be displayed when an associated CourtType is selected if you look at the typeContent.njk it will make things clear

Reply:
split out the regex validation to make the method more easy to follow